lib/clowder-common-ruby/rails_config.rb

```diff
@@ -0,0 +1,175 @@
+        else
+          raise "Unsupported Kafka security protocol '#{security_protocol}'"
+        end
+      else
+        kafka_config[:security_protocol] = 'plaintext'
+      end
+      
+      return kafka_config
+    end
+    
+    # List of Kafka brokers
+    def kafka_brokers
+      first_config.kafka.brokers&.map do |broker|
+        "#{broker.hostname}:#{broker.port}"
+      end&.join(',')
+    end
+
+    # Map for Kafka topics
+    def configure_kafka_topics
+      kafka_topics = {}
+      first_config.kafka.topics&.map do |topic|
+        k = "#{topic.name.sub(/^platform./, '')}".parameterize.underscore.to_sym
+        v = topic.name
+        kafka_topics[k] = v
+      end
+
+      return kafka_topics
+    end
+
+    # Map for redis configuration
+    def configure_redis
+      inMemoryDb = first_config.dig('inMemoryDb')
+      redis_url = "redis://#{inMemoryDb.dig('hostname')}:#{inMemoryDb.dig('port')}"
+
+      redis_config = {
+        url: redis_url,
+        password: inMemoryDb.dig('password')
+      }
+    end
+
+    # Nested map for Cloudwatch configuration
+    def configure_cloudwatch
+      cloudwatch = first_config.logging&.cloudwatch
+      
+      final_cloudwatch_config = {
+        region: cloudwatch&.region,
+        log_group: cloudwatch&.logGroup,
+        log_stream: Socket.gethostname,
+        type: first_config.logging&.type,
+        credentials: {
+          access_key_id: cloudwatch&.accessKeyId,
+          secret_access_key: cloudwatch&.secretAccessKey
+        }
+      }
+    end  
+
+    # Map for RBAC configuration 
+    def configure_rbac
+      rbac_config = first_config.dependency_endpoints.dig('rbac', 'service')
+
+      if first_config.tlsCAPath
+        rbac_host = "#{rbac_config.hostname}:#{rbac_config.tlsPort}"
+        rbac_scheme = 'https'
+      else
+        rbac_host = "#{rbac_config.hostname}:#{rbac_config.port}"
+        rbac_scheme = 'http'
+      end
+
+      final_rbac_config = {
+        host: rbac_host,
+        scheme: rbac_scheme
+      }
+    end
+
+    # Map for endpoint configuration 
+    def configure_endpoint(endpoint)
+      if first_config.tlsCAPath
+        scheme = 'https'
+        host = "#{endpoint.hostname}:#{endpoint.tlsPort}"
+        url = "#{scheme}://#{host}"
+      else
+        scheme = 'http'
+        host = "#{endpoint.hostname}:#{endpoint.port}"
+        url = "#{scheme}://#{host}"
+      end
+      
+      endpoint_config = {
+        host: host,
+        scheme: scheme,
+        url: url
+      }
+    end
+
+    # Nested map for Rails configuration
+    def self.to_hash
+      rails_config = new
+      first_config = rails_config.first_config
+
+      # Kafka
+      kafka_server_config = rails_config.configure_kafka
+
+      # Cloudwatch
+      cloudwatch_config = rails_config.configure_cloudwatch
+
+      # RBAC
+      rbac_config = rails_config.configure_rbac
+
+      # Redis
+      redis_config = rails_config.configure_redis
```

Comment:
I am not sure about these allocations, you could definitely move these method calls to the right side of the hash values below. Also regarding the `rails_config`, what about not having it at all? Just move all methods as singleton methods, taking one argument (the first config) that they actually need.

```
  first_config = ClowderCommonRuby::Config.load
  clowder_config = {
    kafka: configure_kafka(first_config),
    ...
```